### DISCUSSION TITLE
Support for CSS selectors and positions in `textref`

### DISCUSSION THREAD

**Discussion by HadrienGardeur** [readium/guided-navigation#6] [General]
Now that the Guided Navigation syntax is also being considered for read aloud (see #4), we should also consider adding support for two other fragment types on `textref` IMO:

- CSS selectors using `#css()`
- and/or positions using `#position=X`

When working with text, these additional fragments are not necessarily meant for locating text fragment precisely but can help to disambiguate between two similar text fragments.

To support a read aloud feature, a reading application would generate a Guided Navigation Document where `text` contains the text that will be passed to the TTS engine (although it will most likely be pre-processed first, for example to break it down into utterances) and `textref` would reference the source HTML file and the fragment that should be highlighted as the text is being read.

This would also open the door to textual/audio alignment after an ebook has been authored, for example automatic alignment between an audiobook and an ebook.

**Comment by mickael-menu** [comment]
I agree that `#css()` is a must and will be very useful.

For `#position=x`, I'm not sure we can really use it as we don't have accurate position information when processing the HTML content. As the Readium `position` is computed from the raw bytes length and not character count.
